### ISSUE TITLE
Mesh utilities

### ISSUE THREAD

**Issue by neurolabusc** [niivue/niivue#1028]
A list of upcoming mesh features. Some will be added directly to the NiiVue core project, with other components included in small stand alone examples.

Meshifying voxels. Our [nii2mesh](https://github.com/neurolabusc/nii2mesh) illustrates these as a command line application as well as a [web assembly](https://rordenlab.github.io/nii2meshWeb/) plug in. We will bring these features to NiiVue and pure JavaScript.

 - Converting voxels to meshes using [marching cubes](https://paulbourke.net/geometry/polygonise/) using [Will Usher's](https://github.com/Twinklebear/webgl-marching-cubes/blob/master/js/marching-cubes.js) JavaScript.
 - Option to only extract the [largest cluster](https://github.com/neurolabusc/nii2mesh). This can use the [bwlabel](https://github.com/neuroneural/brainchop/blob/master/bwlabels.js) JavaScript we created for brainchop.
 - [Bubble fill](https://github.com/neurolabusc/nii2mesh) option.
 - [Quadric mesh simplification](https://github.com/sp4cerat/Fast-Quadric-Mesh-Simplification), creating a native JavaScript port.
 - Save meshes. At the moment, NiiVue can save voxel-based volumes but not triangulated meshes. Add support for saving mz3, obj and stl formats.
 - A basic wireframe shader to visualize mesh complexity. Note that WebGL2 does not support geometry shaders, so we can not use the approach of Surfice. High quality WebGL solutions add [barycentric coordinates](https://3d.nih.gov/entries/3dpx-017206) to each vertex. However, this prevents vertex-reuse which places a big burden on the GPU. Therefore, we will use an efficient, simple but crude method like the [NIH 3D viewer](https://3d.nih.gov/entries/3dpx-017206).
 - Extend a mesh live demo to illustrate how a developer can easily [decimate hierarchical meshes](https://github.com/niivue/niivue/issues/1007).

**Comment by neurolabusc** [comment]
Note that the `Add Layer` button in the [Load Mesh Layers live demo](https://niivue.github.io/niivue/features/mesh.loader.html) is not functional. The [current code is missing the push command](https://github.com/niivue/niivue/blob/89d8a01d85ad14d33564cdfa4e8c462ecdb346ff/playwright/e2e/test.nvmeshloaders.readLayer.spec.js#L50):
```
nv.meshes[0].layers.push(meshLayer)
```

**Comment by neurolabusc** [comment]
Once the upcoming PR is merged, the [annot mesh atlases](https://niivue.github.io/niivue/features/mesh.atlas.html) live demo will include a `decimate` pull down menu that demonstrate how to [down sample hierarchical FreeSurfer meshes](https://brainder.org/2016/05/31/downsampling-decimating-a-brain-surface/). Other forms of meshes can be simplified using our [separate simplification methods](https://github.com/niivue/niivue-mesh).

<img width="520" alt="decimate" src="https://github.com/user-attachments/assets/9150c918-4242-4c0b-a732-c86dc93dc150">

**Comment by neurolabusc** [comment]
Closing this issue as completed.
 - The `Decimation` pull down menu of [this live demo](https://niivue.github.io/niivue/features/mesh.atlas.html) demonstrates collapse of hierarchical meshes. Vertex colors use nearest neighbor interpolation.
 - The ability to generate meshes from geometry (vertices and triangle indices) and download meshes is [demonstrated in this live demo](https://niivue.github.io/niivue/features/mesh.create.html).
 - Voxel-to-mesh creation as well as simplification is demonstrated in this [github repository](https://github.com/niivue/niivue-mesh) and [this live demo](https://niivue.github.io/niivue-mesh/). Since these features may be useful for ThreeJS and VTK.js developers, I have kept these modular features separate from the core NiiVue project.
